taar/recommenders/ensemble_recommender.py

```diff
@@ -0,0 +1,84 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import logging
+import itertools
+from ..recommenders import utils
+from .base_recommender import BaseRecommender
+
+S3_BUCKET = 'telemetry-parquet'
+ENSEMBLE_WEIGHTS = 'taar/ensemble/ensemble_weight.json'
+
+logger = logging.getLogger(__name__)
+
+
+class EnsembleRecommender(BaseRecommender):
+    """
+    The EnsembleRecommender is a collection of recommenders where the
+    results from each recommendation is amplified or dampened by a
+    factor.  The aggregate results are combines and used to recommend
+    addons for users.
+    """
+    def __init__(self, recommender_map):
+        tmp = utils.get_s3_json_content(S3_BUCKET, ENSEMBLE_WEIGHTS)
+        self._ensemble_weights = tmp['ensemble_weights']
+
+        # Copy the map of the recommenders
+
+        # TODO: verify that the recommender keys match what we've used
+        # in the ensemble training
+        self.RECOMMENDER_KEYS = ['legacy', 'collaborative', 'similarity', 'locale']
+        self._recommender_map = recommender_map
+
+    def can_recommend(self, client_data, extra_data={}):
```

Comment:
If no other "base" recommender is available, then this should return False.

Reply:
Fixed it in  d9fed2beb